src/debug.ts

```diff
@@ -43,20 +52,67 @@ export async function makeDebugConfigurations(ctx: FolderContext) {
+// Return array of DebugConfigurations for executables based on what is in Package.swift
+function createExecutableConfigurations(ctx: FolderContext): vscode.DebugConfiguration[] {
     const executableProducts = ctx.swiftPackage.executableProducts;
 
-    return executableProducts.map((product) => {
-        return {
-            type: "lldb",
-            request: "launch",
-            name: `Run ${product.name}`,
-            program: "${workspaceFolder}/.build/debug/" + product.name,
-            args: [],
-            cwd: "${workspaceFolder}",
-            preLaunchTask: `swift: Build ${product.name}`
-        };    
-    
+    return executableProducts.flatMap((product) => {
+        return [
+            {
+                type: "lldb",
+                request: "launch",
+                name: `Debug ${product.name}`,
+                program: "${workspaceFolder}/.build/debug/" + product.name,
+                args: [],
+                cwd: `\${workspaceFolder:${ctx.folder.name}}`,
+                preLaunchTask: `swift: Build Debug ${product.name}`
+            },
+            {
+                type: "lldb",
+                request: "launch",
+                name: `Release ${product.name}`,
+                program: "${workspaceFolder}/.build/release/" + product.name,
+                args: [],
+                cwd: `\${workspaceFolder:${ctx.folder.name}}`,
+                preLaunchTask: `swift: Build Release ${product.name}`
+            }
+        ];    
     });
 }
+
+// Return array of DebugConfigurations for tests based on what is in Package.swift
+async function createTestConfigurations(ctx: FolderContext): Promise<vscode.DebugConfiguration[]> {
+    if (ctx.swiftPackage.getTargets('test').length === 0) { return []; }
+    try {
+        // If running on darwin. Find xctest exe and run pointing at xctest resources
+        if (process.platform === 'darwin') {
+            const xcodePath = await getXcodePath();
+            if (xcodePath === undefined) {
+                return [];
+            }
+            return [{
+                type: "lldb",
+                request: "launch",
+                name: `Test ${ctx.swiftPackage.name}`,
+                program: `${xcodePath}/usr/bin/xctest`,
+                args: [`.build/debug/${ctx.swiftPackage.name}PackageTests.xctest`],
+                cwd: `\${workspaceFolder:${ctx.folder.name}}`,
+                preLaunchTask: `swift: Build All`
+            }];
+        } else {
+            // otherwise run xctest exe inside build folder
+            return [{
+                type: "lldb",
+                request: "launch",
+                name: `Test ${ctx.swiftPackage.name}`,
+                program: `./.build/debug/${ctx.swiftPackage.name}PackageTests.xctest`,
+                cwd: `\${workspaceFolder:${ctx.folder.name}}`,
+                preLaunchTask: `swift: Build All`
+            }];
+        }
+    } catch {
```

Comment:
> I've checked that the **.xctest** file is present after compilation, but it's not an executable.

Great so Windows do it differently to Linux. I guess we'll have to find out how we can trigger tests correctly on Windows. Temporarily I'll disable

Reply:
getXcodePath can throw I believe

Reply:
It shouldn't throw I think, because you catch any errors and return `undefined` instead? If I understand JavaScript async/await correctly, that `undefined` should be a resolved promise, no?

Reply:
I haven't got the code in front of me at the moment. I'll look at it tomorrow. You're probably right

Reply:
ok temporarily disabled for win32 until we work out what to do, also removed the unnecessary try catch